lib/CdeContext.tsx

```diff
@@ -0,0 +1,67 @@
+import {createContext, PropsWithChildren, useContext, useState} from 'react';
+import {Config, DatasetCDEMapping, InitParams, STEPS} from "./models.ts";
+import theme from "./theme.ts";
+import {ThemeProvider} from "@mui/material";
+
+export const CdeContext = createContext({
+    labName: '',
+    config: {
+        width: "100%",
+        height: "100%",
+    } as Config,
+    cdeFileMapping: null as File | null,
+    mapping: {} as DatasetCDEMapping,
+    setMapping: (_mapping: DatasetCDEMapping) => {},
+    step: 0,
+    setStep: (_step: number) => {},
+    loadingMessage: null as string | null,
+    setLoadingMessage: (_loadingMessage: string | null) => {},
+    errorMessage: null as string | null,
+    setErrorMessage: (_errorMessage: string | null) => {},
+    isOpen: true,
+    handleClose: () => {}
+});
+
+export const useCdeContext = () => useContext(CdeContext);
+
+export const CdeContextProvider = ({children, labName, callback, cdeFileMapping, config}: PropsWithChildren<InitParams>) => {
```

Comment:
This is the 'real' entry point of our library. Right now it's responsible to create the application context and theme. Anything from this point on is like a normal application like we have been developing